File: app/src/main/java/com/apps/kunalfarmah/furnituremagic/Room/FurnitureDao.java
package com.apps.kunalfarmah.furnituremagic.Room;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface FurnitureDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Furniture item);

    @Query("SELECT * from FURNITURE ORDER BY ID ASC")
    LiveData<List<Furniture>> getAllFurnitures();

    @Query("DELETE FROM FURNITURE")
    void deleteAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertFurnitures(List<Furniture> items);

}

File: app/src/main/java/com/apps/kunalfarmah/furnituremagic/Room/FurnitureDao.kt
package com.apps.kunalfarmah.furnituremagic.Room

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface FurnitureDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(item: Furniture?)

    @get:Query("SELECT * from FURNITURE ORDER BY ID ASC")
    val allFurnitures: LiveData<List<Furniture?>?>

    @Query("DELETE FROM FURNITURE")
    fun deleteAll()

    @Delete()
    fun delete(item: Furniture?)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertFurnitures(items: List<Furniture?>?)
}